src/mixins/java/org/spongepowered/common/mixin/api/mcp/world/IWorldMixin_API.java

```diff
@@ -191,158 +218,179 @@ default WorldProperties getProperties() {
-        return null;
-    }
+        // TODO - replace this with an actual check
+        /*
+        if (entity instanceof EntityHanging) {
+            if (((EntityHanging) entity).facingDirection == null) {
+                // TODO Some sort of detection of a valid direction?
+                // i.e scan immediate blocks for something to attach onto.
+                ((EntityHanging) entity).facingDirection = EnumFacing.NORTH;
+            }
+            if (!((EntityHanging) entity).onValidSurface()) {
+                return Optional.empty();
+            }
+        }*/
+
+        if (naturally && entity instanceof MobEntity) {
+            // Adding the default equipment
+            final DifficultyInstance difficulty = this.shadow$getDifficultyForLocation(new BlockPos(x, y, z));
+            ((MobEntityAccessor)entity).accessor$setEquipmentBasedOnDifficulty(difficulty);
+        }
 
-    @Override
-    default DataTransactionResult copyFrom(final int xTo, final int yTo, final int zTo, final DataHolder from) {
-        return null;
-    }
+        if (entity instanceof PaintingEntity) {
+            // This is default when art is null when reading from NBT, could
+            // choose a random art instead?
+            ((PaintingEntity) entity).art = PaintingType.KEBAB;
+        }
 
-    @Override
-    default DataTransactionResult copyFrom(
-        final int xTo, final int yTo, final int zTo, final DataHolder from, final MergeFunction function) {
-        return null;
+        return (Entity) entity;
     }
 
     @Override
-    default DataTransactionResult copyFrom(final int xTo, final int yTo, final int zTo, final int xFrom, final int yFrom, final int zFrom, final MergeFunction function) {
-        return null;
-    }
+    default Collection<Entity> spawnEntities(final Iterable<? extends Entity> entities) {
+        final List<Entity> entitiesToSpawn = new NonNullArrayList<>();
+        entities.forEach(entitiesToSpawn::add);
+        final SpawnEntityEvent.Custom event = SpongeEventFactory
+                .createSpawnEntityEventCustom(Sponge.getCauseStackManager().getCurrentCause(), entitiesToSpawn);
+        if (Sponge.getEventManager().post(event)) {
+            return ImmutableList.of();
+        }
+        for (final Entity entity : event.getEntities()) {
+            EntityUtil.processEntitySpawn(entity, Optional::empty);
+        }
 
-    @Override
-    default boolean validateRawData(final int x, final int y, final int z, final DataView container) {
-        return false;
+        final ImmutableList.Builder<Entity> builder = ImmutableList.builder();
+        for (final Entity entity : event.getEntities()) {
+            builder.add(entity);
+        }
+        return builder.build();
     }
 
     @Override
-    default void setRawData(final int x, final int y, final int z, final DataView container) throws InvalidDataException {
+    default boolean spawnEntity(Entity entity) {
```

Comment:
@gabizou api7 code was capturing stuff in this method. Is this needed here?
Otherwise `IWorldWriterMixin_API` already has the redirect to the `shadow$addEntity`